uui-core/src/data/processing/views/Tree.ts

```diff
@@ -223,47 +215,71 @@ export class Tree<TItem, TId> {
                 parentSubtotals = add(itemSubtotals, subtotalsMap.get(parentId));
             }
             subtotalsMap.set(parentId, parentSubtotals);
-        }, { direction: 'bottom-up' })
+        }, { direction: 'bottom-up' });
         return subtotalsMap;
     }
 
     public isFlatList() {
         return this.byParentId.size <= 1;
     }
 
-    public append(itemsToAdd: TItem[]) {
-        if (!itemsToAdd || itemsToAdd.length === 0) {
+    public patch(
+        items: TItem[],
+        isDeletedProp?: keyof TItem,
+        comparator?: (newItem: TItem, existingItem: TItem) => number,
```

Comment:
Let's provide a default value for comparator. I suggest something like () => -1 - to insert new items at the top of the list by default.